src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  color: Color;
+  shape?: Shape;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+    public shape?: Shape,
+  ) {
+    this.shape = 'triangle';
```

Comment:
we can use this implementetion in class body, because all triangles will have this propery

Reply:
fixed